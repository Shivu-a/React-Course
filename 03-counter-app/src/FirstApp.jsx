import PropTypes from "prop-types";

function App({ title, subTitle }) {
  return (
    <>
      <h1 data-testid="test-title"> {title} </h1>
      <h2> {subTitle} </h2>
    </>
  );
}

App.propTypes = {
  title: PropTypes.string.isRequired,
};

App.defaultProps = {
  title: "No hay titulo",
  subTitle: "No hay sub-titulo",
};

export default App;
