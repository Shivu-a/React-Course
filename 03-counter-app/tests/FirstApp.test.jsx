import { render } from "@testing-library/react";
import FirstApp from "../src/FirstApp";

describe("Pruebas en FirstApp.jsx", () => {
  test("Debe hacer match con el snapshot", () => {
    const title = "Hola soy Rimuru";
    const { container } = render(<FirstApp title={title} />);

    expect(container).toMatchSnapshot();
  });

  test("Debe de mostrar el titulo en un H1", () => {
    const titulo = "Hola soy Rimuru";
    const { container, getByText } = render(<FirstApp title={titulo} />);

    expect(getByText(titulo)).toBeTruthy;

    const h1 = container.querySelector("h1");

    expect(h1.innerHTML).toContain(titulo);
  });
});
