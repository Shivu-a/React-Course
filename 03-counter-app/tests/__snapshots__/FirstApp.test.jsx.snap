// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp.jsx Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
     
    Hola soy Rimuru
     
  </h1>
  <h2>
     
    No hay sub-titulo
     
  </h2>
</div>
`;
